composables/api/useLoginUser.ts

```diff
@@ -0,0 +1,60 @@
+import { ref } from 'vue'
+import { useRuntimeConfig, useState } from '#app'
+import { LoginRequest, User } from '@/lib/api/data-contracts'
+import { Users } from '@/lib/api/Users'
+
+interface LoginUserResponse {
+  loading: boolean
+  error: string | null
+}
+
+export const useLoginUser = (requestData: LoginRequest): LoginUserResponse => {
+  const responseLoading = ref(true)
+  const responseError = ref<string | null>(null)
+  const authToken = useState<string>('session_token')
+  const currentUser = useState<User>('current_user', () => ({
+    admin: false,
+    email: '',
+    username: '',
+  }))
+
+  const userClient = new Users({
+    baseUrl: useRuntimeConfig().public.BACKEND_BASE_URL,
+  })
+  userClient
+    .usersLoginCreate(requestData)
+    .then((response) => {
+      if (response.status >= 200 && response.status < 300) {
```

Comment:
Nit: we should be able to call `response.ok` here, right?

Reply:
I actually didn't know that existed. TIL 🎉 

[`response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) documentation in case anyone else sees this